src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -163,19 +172,36 @@ public static class PrereleaseCompatibilityTransformer
 
                     if (run["files"] is JObject files)
                     {
-                        keyToIndexMap = new Dictionary<string, int>();
+                        fileKeyToIndexMap = new Dictionary<string, int>();
 
                         run["files"] =
                             ConstructFilesArray(
                                 files,
-                                keyToIndexMap);
+                                fileKeyToIndexMap);
 
-                        modifiedLog |= keyToIndexMap.Count > 0;
+                        modifiedLog |= fileKeyToIndexMap.Count > 0;
                     }
 
+                    if (run["resources"] is JObject resources)
+                    {
+                        // Remove 'open' from rule configuration default level enumeration
+                        // https://github.com/oasis-tcs/sarif-spec/issues/288
+                        modifiedLog |= RemapRuleDefaultLevelFromOpenToNote(resources);
+
+                        ruleKeyToIndexMap = new Dictionary<string, int>();
 
-                    var results = (JArray)run["results"];
-                    if (results != null)
+                        if (resources["rules"] is JObject rules)
+                        {
+                            resources["rules"] =
+                                ConstructRulesArray(
```

Comment:
>ConstructRulesArray [](start = 32, length = 19)

How about `ConvertRulesDictionaryToArray`. I had to do a quick glance upward to remember that variable `rules` is a dictionary at this point.

Reply:
For that matter, same thing for `ConstructFilesArray`.

---
In reply to: [249161968](https://github.com/Microsoft/sarif-sdk/pull/1223#discussion_r249161968) [](ancestors = 249161968)